packages/property-view/src/browser/react-components/App.tsx

```diff
@@ -0,0 +1,49 @@
+/********************************************************************************
+ * Copyright (c) 2023 CrossBreeze.
+ ********************************************************************************/
+
+import * as React from '@theia/core/shared/react';
+import { ErrorView } from './views/ErrorView';
+import { CrossModelRoot } from '@crossbreeze/protocol';
+import { ModelProvider, ModelReducer } from './ModelContext';
+import { EntityPropertyView } from './views/EntityPropertyView';
+import SaveModelContext from './SaveModelContext';
+import _ = require('lodash');
+
+interface AppProperty {
+    model: CrossModelRoot | undefined;
+    saveModel: () => void;
+    updateModel: (model: CrossModelRoot) => void;
+}
+
+export function App(props: AppProperty): React.ReactElement {
+    const [model, dispatch] = React.useReducer(ModelReducer, props.model as CrossModelRoot);
+
+    React.useEffect(() => {
+        dispatch({ type: 'model:update', model: props.model });
+    }, [props.model]);
+
+    React.useEffect(() => {
+        props.updateModel(_.cloneDeep(model));
+    }, [model, props]);
+
+    let content = <></>;
+
+    if (!model) {
+        return <ErrorView errorMessage='No model given to the property view!' />;
```

Comment:
Yeah, it is kind of weird. The model is probably never empty, unless there is a programming error. The reason I put it here is because typescript start complaining if you do not check it. I check it here, so we do not have to check it in every if in the next if else. 

Changed it to return an empty fragment.